nat-handler.py

```diff
@@ -23,14 +23,17 @@
 
         {
             "eu-west-1a": {
+                "secondary_interface_id": "eni-abc123",
                 "route_table_id": "rtb-0e0ed06b",
                 "elastic_ip_allocation_id": "eipalloc-cc618fa9"
             },
             "eu-west-1b": {
+                "secondary_interface_id": "eni-def456",
                 "route_table_id": "rtb-090ed06c",
                 "elastic_ip_allocation_id": "eipalloc-c5618fa0",
             },
             "eu-west-1c": {
+                "secondary_interface_id": "eni-ghi789",
```

Comment:
Yes, it can be derived at runtime, but this config is used by the whaleshark-nat.upstart script. On reboot, the instance needs to attach a secondary interface eth1. Hence for this particular "launch" and "reboot" case, i thought it was simpler to have 3 interfaces already created and then simply attach it to the instance.

Another approach what i was doing previously was to create a network interface on the fly i.e. on reboot. But its possible that the network interface creation fails, and also there might be more work to do on reboot which causes a slower reboot.

Another problem with this approach was that, if we create new network interfaces (eni) on launch and reboot, then new ENIs would be created all the time during a launch / reboot. Then the question is, what do we do the the ENIs already created before? They will be lying around and not in use (and not attached to any instance).

For this, we would then need a mechanism to either remove un-used ENIs or each time we reboot, find an ENI that is not in use. This would then also introduce race-conditions and retries until an instance can attach itself to an ENI while another instance that is also rebooting at the same time is trying to get a hold of the ENI.

So all this seemed very complicating and this solution seemed very simple to implement and understand in terms of implementation.

Hope i've made my point clear :)